code/src/components/ThoughtForm.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import heart from '../assets/heart.png'
+
+const ThoughtForm = ({onFormSubmit, setNewThought, newThought, newName, setNewName}) => {
+    return (
+
+      <form onSubmit={onFormSubmit} className="form-container">
+        <label htmlFor="newThought" className="label-form">What's making you Happy right now?</label>
+        <textarea 
+          className="thought-input"
+          id="newThought"
+          type="text" 
+          value={newThought}
+          onChange= {(e) => setNewThought(e.target.value)}
+          ></textarea>
+          <input
+            className="name-input"
+            type='text' 
+            placeholder='Your name (optional)'
+            maxLength='20'
+            value={newName}
+            onChange={(e) => setNewName(e.target.value)}>
+            </input>
+            
+          <div className="button-count">
+            <button disabled = {newThought.length<5 || newThought.length>140}type="submit" className="submit-button">
```

Comment:
This is nice, disabling the button for minimum and maximum characters and also the character counter. :)